Gemfile

```diff
@@ -5,32 +5,39 @@ gemspec
 # Development dependencies
 gem 'addressable', '~> 2.4.0' # locking transitive dependency of webmock
 gem 'appraisal', '~> 2.2'
-gem 'benchmark-ips', '~> 2.8'
+gem 'benchmark-ips', '~> 2.9'
 gem 'benchmark-memory', '~> 0.1'
-gem 'builder'
+gem 'builder', '>= 2.1.2'
 gem 'climate_control', '~> 0.2.0'
 # Leave it open as we also have it as an integration and want Appraisal to control the version under test.
-gem 'concurrent-ruby'
+gem 'concurrent-ruby', '>= 0.9'
 gem 'memory_profiler', '~> 0.9'
 gem 'minitest', '= 5.10.1'
 gem 'minitest-around', '0.5.0'
 gem 'minitest-stub_any_instance', '1.0.2'
-gem 'pimpmychangelog', '>= 0.1.2'
-gem 'pry'
+gem 'pimpmychangelog', '>= 0.1.3'
+gem 'pry', '>= 0.12.2'
 if RUBY_PLATFORM != 'java'
   # There's a few incompatibilities between pry/pry-byebug on older Rubies
-  gem 'pry-byebug' if RUBY_VERSION >= '2.6.0' && RUBY_ENGINE != 'truffleruby'
-  gem 'pry-nav' if RUBY_VERSION < '2.6.0'
-  gem 'pry-stack_explorer'
+  gem 'pry-byebug', '>= 3.8.0' if RUBY_VERSION >= '2.6.0' && RUBY_ENGINE != 'truffleruby'
+  gem 'pry-nav', '>= 0.3.0' if RUBY_VERSION < '2.6.0'
+  gem 'pry-stack_explorer', '>= 0.4.9.3'
 else
-  gem 'pry-debugger-jruby'
+  gem 'pry-debugger-jruby', '>= 2.0.0'
 end
-gem 'rake', '>= 10.5', '< 13.0.4' # Locking rake until https://github.com/thoughtbot/appraisal/pull/184 is released
-gem 'redcarpet', '~> 3.4' if RUBY_PLATFORM != 'java'
+gem 'rake', '>= 12.0', '< 13.0.4' # Locking rake until https://github.com/thoughtbot/appraisal/pull/184 is released
+gem 'redcarpet', '~> 3.5' if RUBY_PLATFORM != 'java'
 gem 'rspec', '~> 3.10'
-gem 'rspec-collection_matchers', '~> 1.1'
+gem 'rspec-collection_matchers', '~> 1.2'
 gem 'rspec_junit_formatter', '>= 0.4.1'
-gem 'rspec_n', '~> 1.3' if RUBY_VERSION >= '2.3.0'
+if RUBY_VERSION >= '2.3.0'
+  gem 'rspec_n', '~> 1.3'
+
+  # Ancient bundler mistakenly installs incompatible cri version for Ruby 2.3.
+  # We can't update bundler because it would mean going from bundler v1 to v2,
+  # and that breaks many gems that we test with in Ruby 2.3.
+  gem 'cri', '< 2.15.11' if RUBY_VERSION < '2.4.0'
+end
```

Comment:
Instead of this change, I suggest just moving the minimum ruby version for rspec_n to 2.4. (I think I left this comment elsewhere already... deja vu lol?)